src/httpserver.cpp

```diff
@@ -288,11 +282,9 @@ static void http_request_cb(struct evhttp_request* req, void* arg)
 
     // Dispatch to worker thread
     if (i != iend) {
-        std::auto_ptr<HTTPWorkItem> item(new HTTPWorkItem(hreq.release(), path, i->handler));
+        std::unique_ptr<HTTPWorkItem> item(new HTTPWorkItem(hreq.release(), path, i->handler));
         assert(workQueue);
-        if (workQueue->Enqueue(item.get()))
-            item.release(); /* if true, queue took ownership */
-        else
+        if (!workQueue->Enqueue(std::move(item)))
             item->req->WriteReply(HTTP_INTERNAL, "Work queue depth exceeded");
```

Comment:
Leaving out the std::move in the `->Enqueue` line gives a compile error about pointer conversions:
```
/home/orion/projects/bitcoin/bitcoin/src/httpserver.cpp:289:33: error: no viable conversion from 'unique_ptr<HTTPWorkItem>' to 'unique_ptr<HTTPClosure>'
        if (!workQueue->Enqueue(item))
```
Bleh.